File: python/bob/bob.py
import re

def response(hey_bob):
    words = re.findall(r'[a-zA-Z\d\?]', hey_bob)

    if len(words) == 0:
        return 'Fine. Be that way!'
    
    if words[-1] == '?':
        if len(words) > 1 and all(e.isupper() for e in words[:-1]):
            return 'Calm down, I know what I\'m doing!'
        else:
            return 'Sure.'
    else:
        words_without_numbers = re.findall(r'[a-zA-Z]', hey_bob)
        if len(words_without_numbers) > 0 and all(e.isupper() for e in words_without_numbers):
            return 'Whoa, chill out!'
        else:
            return 'Whatever.'

File: typescript/bob/bob.ts
export function hey(message: string): string {
  const responseList: string[] = [
    "Whoa, chill out!",
    "Sure.",
    "Fine. Be that way!",
    "Whatever.",
    "Calm down, I know what I'm doing!",
  ];
  message = message.trim().split("\n").join(" ");
  const characters = message.match(/[a-zA-Z]+/);
  const numbers = message.match(/\d/);
  if (
    (characters === null && numbers === null) ||
    (message && message.length === 0)
  ) {
    return responseList[2];
  } else if (
    characters &&
    characters.length !== 0 &&
    characters.every((x) => isNaN(parseInt(x)) && x === x.toLocaleUpperCase())
  ) {
    if (message.match(/.*\?$/m)) {
      return responseList[4];
    } else {
      return responseList[0];
    }
  } else if (message.match(/.*\?$/m)) {
    return responseList[1];
  }
  return responseList[3];
}
